AWS-VPC_deploy.yaml

```diff
@@ -0,0 +1,214 @@
+
+  PubSub1CidrBlock:
+    Type: String
+    Description: Public Subnet 1 Cidr Block
+    Default: 10.0.1.0/24
+    ConstraintDescription: Must be a valid CIDR block within the VPC CIDR range (e.g., 10.0.1.0/24)
+
+  PubSub2CidrBlock:
+    Type: String
+    Description: Public Subnet 2 Cidr Block
+    Default: 10.0.2.0/24
+    ConstraintDescription: Must be a valid CIDR block within the VPC CIDR range (e.g., 10.0.2.0/24)
+
+  PrivSub1CidrBlock:
+    Type: String
+    Description: Private Subnet 1 Cidr Block
+    Default: 10.0.3.0/24
+    ConstraintDescription: Must be a valid CIDR block within the VPC CIDR range (e.g., 10.0.3.0/24)
+
+  PrivSub2CidrBlock:
+    Type: String
+    Description: Private Subnet 2 Cidr Block
+    Default: 10.0.4.0/24
+    ConstraintDescription: Must be a valid CIDR block within the VPC CIDR range (e.g., 10.0.4.0/24)
+
+
+Resources:
+
+# Creating VPC
+  VPC:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: !Ref VPCCidrBlock
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: AWS-VPC
+
+# VPC Internet Gateway
+  InternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: InternetGateway
+
+# Attaching VPC Internet Gateway
+  VPCGatewayAttachment:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId: !Ref VPC
+      InternetGatewayId: !Ref InternetGateway
+
+# Creating Subnets, 2 Pub, 2 Priv
+  PublicSubnet1:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      CidrBlock: !Ref PubSub1CidrBlock
+      AvailabilityZone: !Select [0, !GetAZs ""]
+      MapPublicIpOnLaunch: true
+      Tags:
+        - Key: Name
+          Value: PublicSubnet1
+
+  PublicSubnet2:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      CidrBlock: !Ref PubSub2CidrBlock
+      AvailabilityZone: !Select [1, !GetAZs ""]
+      MapPublicIpOnLaunch: true
+      Tags:
+        - Key: Name
+          Value: PublicSubnet2
+
+  PrivateSubnet1:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      CidrBlock: !Ref PrivSub1CidrBlock
+      AvailabilityZone: !Select [0, !GetAZs ""]
+      Tags:
+        - Key: Name
+          Value: PrivateSubnet1
+
+  PrivateSubnet2:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      CidrBlock: !Ref PrivSub2CidrBlock
+      AvailabilityZone: !Select [1, !GetAZs ""]
+      Tags:
+        - Key: Name
+          Value: PrivateSubnet2
+
+# Creating 1 route table for both public subnets
+  PublicRouteTable:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref VPC
+      Tags:
+        - Key: Name
+          Value: PublicRouteTable
+
+# Creating internet route and associating it to the InternetGateway
+  DefaultPublicInternetRoute:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref PublicRouteTable
```

Comment:
You Ref to another component while you are not sure it is generated or no. Use DependOn attribute to solve this problem

Reply:
This comment is general for all components that depend on other ones

Reply:
added DependsOns